sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -962,16 +971,25 @@ private Progress getProgress() {
             .build());
   }
 
-  private <K> void processTimer(String timerId, TimeDomain timeDomain, Timer<K> timer) {
+  private <K> void processTimer(
+      String timerIdOrTimerFamilyId, TimeDomain timeDomain, Timer<K> timer) {
     currentTimer = timer;
     currentTimeDomain = timeDomain;
     onTimerContext = new OnTimerContext<>(timer.getUserKey());
+    String timerId =
+        timerIdOrTimerFamilyId.startsWith(TimerFamilyDeclaration.PREFIX)
```

Comment:
If the `timerIdOrTimerFamilyId ` is for a timer family, should the timerId be the `timer.dynamicTimerTag`?

Reply:
it will be ignored anyway, apparently only one of timerId or the timerFamilyId takes effect.
https://github.com/apache/beam/blob/591de3473144de54beef0932131025e2a4d8504b/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java#L223